### PR TITLE
A very brief explanation of how to add custom functions to OpenSSL.

### PR DESCRIPTION
**Pull request by beldmit**
Inspired by Rich Salz's letter to openssl-users@

##### Checklist

- [x] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by InfoHunter** [comment]
Should `make test` be mentioned as well?

File: HACKING [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ If you are adding new public functions to the custom library build, you need to
+ either add a prototype in one of the existing OpenSSL header files;
+ or provide a new header file and edit Configurations/unix-Makefile.tmpl to pick up that file.
+ 
+ After that perform the following steps:
+ 
+    ./configure '-Werror --strict-warnings' [your-options]
+    make update
+    make
+		make test
```
Comment (InfoHunter) [inline HACKING]: Indents seem not correct (Tabs instead of spaces)

File: HACKING [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ ------------------------
+ This document describes the way to add custom modifications to OpenSSL sources.
+ 
+ If you are adding new public functions to the custom library build, you need to
+ either add a prototype in one of the existing OpenSSL header files;
+ or provide a new header file and edit Configurations/unix-Makefile.tmpl to pick up that file.
+ 
+ After that perform the following steps:
+ 
+    ./configure '-Werror --strict-warnings' [your-options]
```
Comment (paulidale) [inline HACKING]: This still works without the single quotes, best to be rid of them.

Comment (romen) [reply HACKING]: Also, it should be either config or Configure, but not configure

Comment (paulidale) [reply HACKING]: Now, I feel like I had selective blindness.  Best make it:
```
./config -Werror --strict-warnings [your-options]
```

Comment (beldmit) [reply HACKING]: Fixed.

**Comment by levitte** [comment]
You gonna merge this, @paulidale?

**Comment by paulidale** [comment]
I've merged but @beldmit is a committer...

**Pull request closed without merging**